be/src/exec/json_scanner.cpp

```diff
@@ -0,0 +1,479 @@
+    }
+    case TFileType::FILE_BROKER: {
+        BrokerReader* broker_reader = new BrokerReader(
+            _state->exec_env(), _broker_addresses, _params.properties, range.path, start_offset);
+        RETURN_IF_ERROR(broker_reader->open());
+        file = broker_reader;
+        break;
+    }
+
+    case TFileType::FILE_STREAM: {
+        _stream_load_pipe = _state->exec_env()->load_stream_mgr()->get(range.load_id);
+        if (_stream_load_pipe == nullptr) {
+            VLOG(3) << "unknown stream load id: " << UniqueId(range.load_id);
+            return Status::InternalError("unknown stream load id");
+        }
+        file = _stream_load_pipe.get();
+        break;
+    }
+    default: {
+        std::stringstream ss;
+        ss << "Unknown file type, type=" << range.file_type;
+        return Status::InternalError(ss.str());
+    }
+    }
+
+    std::string jsonpath = "";
+    std::string jsonpath_file = "";
+    if (range.__isset.jsonpath) {
+        jsonpath = range.jsonpath;
+    } else if (range.__isset.jsonpath_file) {
+        jsonpath_file = range.jsonpath_file;
+    }
+    _cur_file_reader = new JsonReader(_state->exec_env()->small_file_mgr(), _profile, file, jsonpath, jsonpath_file);
+
+    return Status::OK();
+}
+
+void JsonScanner::close() {
+    if (_cur_file_reader != nullptr) {
+        if (_stream_load_pipe != nullptr) {
+            _stream_load_pipe.reset();
+            _cur_file_reader = nullptr;
+        } else {
+            delete _cur_file_reader;
+            _cur_file_reader = nullptr;
+        }
+    }
+}
+
+////// class JsonDataInternal
+JsonDataInternal::JsonDataInternal(rapidjson::Value* v) :
+         _jsonValues(v), _iterator(v->Begin()) {
+}
+
+JsonDataInternal::~JsonDataInternal() {
+
+}
+bool JsonDataInternal::isEnd() {
+    return _jsonValues->End() == _iterator;
+}
+
+rapidjson::Value::ConstValueIterator JsonDataInternal::getNext() {
+    if (isEnd()) {
+        return nullptr;
+    }
+    return _iterator++;
+}
+
+
+////// class JsonReader
+JsonReader::JsonReader(
+        SmallFileMgr *fileMgr,
+        RuntimeProfile* profile,
+        FileReader* file_reader,
+        std::string& jsonpath,
+        std::string& jsonpath_file) :
+            _next_line(0),
+            _total_lines(0),
+            _profile(profile),
+            _file_reader(file_reader) {
+    _bytes_read_counter = ADD_COUNTER(_profile, "BytesRead", TUnit::BYTES);
+    _read_timer = ADD_TIMER(_profile, "FileReadTime");
+
+    //parse jsonpath
+    if (!jsonpath.empty()) {
+        if (!_jsonPathDoc.Parse(jsonpath.c_str()).HasParseError()) {
+            if (!_jsonPathDoc.HasMember("jsonpath") || !_jsonPathDoc["jsonpath"].IsArray()) {
+                _parseJsonPathFlag = -1;// failed, has none object
+            } else {
+                _parseJsonPathFlag = 1;// success
+            }
+        } else {
+            _parseJsonPathFlag = -1;// parse failed
+        }
+    } else if (!jsonpath_file.empty()) {
```

Comment:
why not parse jsonpath in frontend, and pass the json path to backend?

Reply:
I think paese jsonpath in backend, it's easy